src/pivot-table/data/helpers/get-expression-color.ts

```diff
@@ -0,0 +1,23 @@
+import { COLORING, isDarkColor } from "@qlik/nebula-table-utils/lib/utils";
+import type { AttrExprInfoIndex } from "../../../types/types";
+import toRGB from "./to-rgb";
+
+const getExpressionColor = (
+  attrsExprInfo: AttrExprInfoIndex,
+  cell: EngineAPI.INxPivotValuePoint | EngineAPI.INxPivotDimensionCell,
+) => {
+  const attrsExprValues = cell.qAttrExps as unknown as EngineAPI.INxAttributeExpressionValues;
```

Comment:
The type definition for `cell.qAttrExps` is `EngineAPI.INxAttributeExpressionValues | EngineAPI.INxAttributeExpressionValues[]`. This just cast it to the object. Otherwise typescript will complain because `qValues` does not exist on an array.

Reply:
Not really sure when it would be an array. I've never seen it.

Reply:
wow that is a pretty bad union, either it is x or an array of x... potentially leave a comment here, since it looks like pretty risky casting